A tiny JS world - finished

KurosavaAkira opened the pull request:
### A Tiny JS World - [Code base](https://github.com/KurosavaAkira/kottans-frontend/blob/master/task_js-pre-oop/index.js) | [demo](https://kurosavaakira.github.io/kottans-frontend/task_js-pre-oop)

A tiny JS world - finished. Please review the code base in index.js and merge my pull request.

OleksiyRudenko commented:
@KurosavaAkira , good job!

But before we proceed with this PR, please, edit your PR comment and add the following:
- link to the webpage with your showcase so the reviewers could quickly check your web app
- link to the repo where your project originally resides;
  In your case it is not a standalone repo so include the path to the code base in question
- add some comment describing the purpose of this PR for history and for repo maintainers could easily understand what this PR is about before they take a look at changes

You know that the only code to review resides in `index.js` and others may not know this. Pointing this out will help them to focus on that single file and not spending efforts on icons, css, html et al that aren't essential for the code review.

It is always good to take care of your colleagues and give them some specific context.

KurosavaAkira commented:
@OleksiyRudenko unnecessary files for code review has been removed with the last commit.

OleksiyRudenko commented:
@KurosavaAkira best ever! Thanks!

OleksiyRudenko commented on submissions/KurosavaAkira/task_js-pre-oop/README.md:
```diff
@@ -0,0 +1,5 @@
+
+## Building a Tiny JS Worldl
+
+### [A Tiny JS World - demo](https://kurosavaakira.github.io/kottans-frontend/task_js-pre-oop)
```
One somebody may want to take a look at full code base.
Consider adding a link to complete code base.

Not critical, not a blocker. Change only if decide to change anything else in this PR.

KurosavaAkira replied:
Okay

OleksiyRudenko commented on submissions/KurosavaAkira/task_js-pre-oop/index.js:
```diff
@@ -0,0 +1,57 @@
+let message = (obj) => {
+    let string = '';
+    for (var key in obj) {
+        if (obj[key] != 0) {
```
You are not required to fix it now, but think on how you could improve your code using some advanced JS feature.

Things worth mastering:
 - Array.join()
 - destructuring assignments
 - Object.keys()
 - Array.filter()
 - switch operator

KurosavaAkira replied:
Okay, I will

KurosavaAkira commented on submissions/KurosavaAkira/task_js-pre-oop/index.js:
```diff
@@ -0,0 +1,57 @@
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Species {
```
Got it

KurosavaAkira commented on submissions/KurosavaAkira/task_js-pre-oop/index.js:
```diff
@@ -0,0 +1,57 @@
+const man = new Species('man', 'Sebastian', 'This us Kottaaans!', 'male', '2', '2');
+const woman = new Species('woman', 'Eve', 'I like Earth', 'female', '2', '2');
+
+let message = (obj) => {
```
OMG, I made my life harder and I enjoy it. Call the doctor.

KurosavaAkira commented:
@OleksiyRudenko In last commit I fixed naming confusion in class and added a link to the code base in readme.md . Please, review if everything is ok for now.

OleksiyRudenko commented on submissions/KurosavaAkira/task_js-pre-oop/index.js:
```diff
@@ -0,0 +1,57 @@
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Inhabitants {
```
Normally, class name is given in singular form. Since `new ClassName()` creates single instance of a class.

You might have been confused by _species_, which is both singular and plural form of the same word.

You are not required to fix this. Just keep in mind.

OleksiyRudenko commented on submissions/KurosavaAkira/task_js-pre-oop/index.js:
```diff
@@ -0,0 +1,57 @@
+
+const dog = new Inhabitants('dog', 'Rvach', 'woof-woof', 'male', '4');
+const cat = new Inhabitants('cat', 'Markiz', 'NONONONONONONO', 'male', '4');
+const man = new Inhabitants('man', 'Sebastian', 'This us Kottaaans!', 'male', '2', '2');
```
Woman is not a species. Man is not as well. These are generic names do denote gender difference within the same species. The tech specs required to specify attribution to a species.

Also refer to the [past comment on the same topic](https://github.com/kottans/frontend-2019-homeworks/pull/6#discussion_r230029424)

Please, fix.

Should you need further details on the topic, please, let me know.

KurosavaAkira replied:
@OleksiyRudenko  sorry, but I don't really understand what I should fix. I thought that since the class name was renamed to "Inhabitants" it should meet all the requirements in the correct name for all inhabitants including humans and animals.

Could you please tell more what attribution to a species do you mean? Unfortunately, I don't know English so well

OleksiyRudenko replied:
Let's put things straightforward.

Your constructor is OK. Species, name, phrase etc are all properties of an inhabitant. You've got the concept of properties perfectly well.

The point is that species is something gender neutral, e.g. dogs, cats, humans etc.

So at instantiation just replace 'man' and 'woman' with 'human'.

Let me know when you're done. We gonna then merge this PR.

KurosavaAkira replied:
@OleksiyRudenko 'man' and 'woman' has been replaced with 'human' in the last commit.

OleksiyRudenko commented:
@KurosavaAkira Great job! Hope you've learned something extra during this review.
